### ISSUE TITLE
Measurment: SEGV Changing pad length with measurement tool open

### ISSUE THREAD

**Issue by j8spencer** [FreeCAD/FreeCAD#16744]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

I had the measurement tool open, set to length, and measured an edge along a pad. I realized I need that pad longer, so I went to increase the length--selected the pad in the model view, typed in a new length in the parameter text box, hit enter. The application crashed with the following on the console (some of which may have been older):

```
<Sketch> SketchObject.cpp(9053): Failed to project external geometry in BaseSideWall#Sketch001: Binder.;#52:3;:G;CHF;:H9af:7,F;:H9b0,F
Selected external reference plane must be normal to sketch plane
Traceback (most recent call last):
  File "<string>", line 1, in <module>
<class 'ValueError'>: Not able to add external shape element Face1
App.getDocument('BaseSideWall').getObject('Sketch001').addExternal("Binder","Face1")
Failed to add external geometry
<Sketch> SketchObject.cpp(9053): Failed to project external geometry in BaseSideWall#Sketch001: Binder.;#52:3;:G;CHF;:H9af:7,F;:H9b0,F
Selected external reference plane must be normal to sketch plane
Traceback (most recent call last):
  File "<string>", line 1, in <module>
<class 'ValueError'>: Not able to add external shape element Face1
App.getDocument('BaseSideWall').getObject('Sketch001').addExternal("Binder","Face1")
Failed to add external geometry
Link to Edge5 already exists in this sketch.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
<class 'ValueError'>: Not able to add external shape element Edge5
App.getDocument('BaseSideWall').getObject('Sketch001').addExternal("Binder","Edge5")
Failed to add external geometry
<Sketch> SketchObject.cpp(9053): Failed to project external geometry in BaseSideWall#Sketch001: Binder.;#52:3;:G;CHF;:H9af:7,F;:H9b0,F
Selected external reference plane must be normal to sketch plane
Link to Edge5 already exists in this sketch.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
<class 'ValueError'>: Not able to add external shape element Edge5
App.getDocument('BaseSideWall').getObject('Sketch001').addExternal("Binder","Edge5")
Failed to add external geometry
Link to Edge5 already exists in this sketch.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
<class 'ValueError'>: Not able to add external shape element Edge5
App.getDocument('BaseSideWall').getObject('Sketch001').addExternal("Binder","Edge5")
Failed to add external geometry
Link to Edge5 already exists in this sketch.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
<class 'ValueError'>: Not able to add external shape element Edge5
App.getDocument('BaseSideWall').getObject('Sketch001').addExternal("Binder","Edge5")
Failed to add external geometry
Link to Edge5 already exists in this sketch.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
<class 'ValueError'>: Not able to add external shape element Edge5
App.getDocument('BaseSideWall').getObject('Sketch001').addExternal("Binder","Edge5")
Failed to add external geometry
1.25e-07 Selection.cpp(157): Unhandled std::exception caught in selection observer: vector::_M_range_check: __n (which is 0) >= this->size() (which is 0)
Traceback (most recent call last):
  File "/tmp/.mount_FreeCAZXcM2y/usr/Mod/Measure/MeasureCOM.py", line 114, in parseSelection
    obj.Element = (o, item["subName"])
    ^^^^^^^^^^^
<class 'ValueError'>: PropertyLinkSub does not support external object
Traceback (most recent call last):
  File "/tmp/.mount_FreeCAZXcM2y/usr/Mod/Measure/MeasureCOM.py", line 112, in parseSelection
    item = selection[0]
           ~~~~~~~~~^^^
<class 'IndexError'>: tuple index out of range
Traceback (most recent call last):
  File "/tmp/.mount_FreeCAZXcM2y/usr/Mod/Measure/MeasureCOM.py", line 114, in parseSelection
    obj.Element = (o, item["subName"])
    ^^^^^^^^^^^
<class 'ValueError'>: PropertyLinkSub does not support external object
Traceback (most recent call last):
  File "/tmp/.mount_FreeCAZXcM2y/usr/Mod/Measure/MeasureCOM.py", line 112, in parseSelection
    item = selection[0]
           ~~~~~~~~~^^^
<class 'IndexError'>: tuple index out of range
Traceback (most recent call last):
  File "/tmp/.mount_FreeCAZXcM2y/usr/Mod/Measure/MeasureCOM.py", line 114, in parseSelection
    obj.Element = (o, item["subName"])
    ^^^^^^^^^^^
<class 'ValueError'>: PropertyLinkSub does not support external object
Traceback (most recent call last):
  File "/tmp/.mount_FreeCAZXcM2y/usr/Mod/Measure/MeasureCOM.py", line 112, in parseSelection
    item = selection[0]
           ~~~~~~~~~^^^
<class 'IndexError'>: tuple index out of range
Traceback (most recent call last):
  File "/tmp/.mount_FreeCAZXcM2y/usr/Mod/Measure/MeasureCOM.py", line 114, in parseSelection
    obj.Element = (o, item["subName"])
    ^^^^^^^^^^^
<class 'ValueError'>: PropertyLinkSub does not support external object
Traceback (most recent call last):
  File "/tmp/.mount_FreeCAZXcM2y/usr/Mod/Measure/MeasureCOM.py", line 112, in parseSelection
    item = selection[0]
           ~~~~~~~~~^^^
<class 'IndexError'>: tuple index out of range
Traceback (most recent call last):
  File "/tmp/.mount_FreeCAZXcM2y/usr/Mod/Measure/MeasureCOM.py", line 114, in parseSelection
    obj.Element = (o, item["subName"])
    ^^^^^^^^^^^
<class 'ValueError'>: PropertyLinkSub does not support external object
73.1215 Selection.cpp(157): Unhandled std::exception caught in selection observer: vector::_M_range_check: __n (which is 1) >= this->size() (which is 1)
Unhandled std::exception caught in GUIApplication::notify.
The error message is: vector::_M_range_check: __n (which is 1) >= this->size() (which is 1)
Unhandled std::exception caught in GUIApplication::notify.
The error message is: vector::_M_range_check: __n (which is 1) >= this->size() (which is 1)
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e13 missing reference: Binder.;#5c:3;:U;CHF;:H9af:7,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e14 missing reference: Binder.;#5c:3;:U5;CHF;:H9af:8,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e15 missing reference: Binder.;#5c:3;:U6;CHF;:H9af:8,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e16 missing reference: Binder.;#5c:3;:U4;CHF;:H9af:8,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e17 missing reference: Binder.;#5c:3;:U2;CHF;:H9af:8,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e18 missing reference: Binder.;#5c:3;:U3;CHF;:H9af:8,E;:H9b0,E
Missing external geometry reference
Clear constraints to external geometry
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e13 missing reference: Binder.;#5c:3;:U;CHF;:H9af:7,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e14 missing reference: Binder.;#5c:3;:U5;CHF;:H9af:8,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e15 missing reference: Binder.;#5c:3;:U6;CHF;:H9af:8,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e16 missing reference: Binder.;#5c:3;:U4;CHF;:H9af:8,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e17 missing reference: Binder.;#5c:3;:U2;CHF;:H9af:8,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e18 missing reference: Binder.;#5c:3;:U3;CHF;:H9af:8,E;:H9b0,E
Missing external geometry reference
Clear constraints to external geometry
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e13 missing reference: Binder.;#5c:3;:U;CHF;:H9af:7,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e14 missing reference: Binder.;#5c:3;:U5;CHF;:H9af:8,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e15 missing reference: Binder.;#5c:3;:U6;CHF;:H9af:8,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e16 missing reference: Binder.;#5c:3;:U4;CHF;:H9af:8,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e18 missing reference: Binder.;#5c:3;:U3;CHF;:H9af:8,E;:H9b0,E
Missing external geometry reference
Clear constraints to external geometry
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e14 missing reference: Binder.;#5c:3;:U5;CHF;:H9af:8,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e15 missing reference: Binder.;#5c:3;:U6;CHF;:H9af:8,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e16 missing reference: Binder.;#5c:3;:U4;CHF;:H9af:8,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e18 missing reference: Binder.;#5c:3;:U3;CHF;:H9af:8,E;:H9b0,E
Missing external geometry reference
Clear constraints to external geometry
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e14 missing reference: Binder.;#5c:3;:U5;CHF;:H9af:8,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e15 missing reference: Binder.;#5c:3;:U6;CHF;:H9af:8,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e18 missing reference: Binder.;#5c:3;:U3;CHF;:H9af:8,E;:H9b0,E
Missing external geometry reference
Clear constraints to external geometry
Traceback (most recent call last):
  File "<string>", line 1, in <module>
<class 'ValueError'>: Datum 0.375 in for the constraint with index 13 is invalid
App.getDocument('BaseSideWall').getObject('Sketch001').setDatum(13,App.Units.Quantity('9.525000 mm'))
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e14 missing reference: Binder.;#5c:3;:U5;CHF;:H9af:8,E;:H9b0,E
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e15 missing reference: Binder.;#5c:3;:U6;CHF;:H9af:8,E;:H9b0,E
Missing external geometry reference
Clear constraints to external geometry
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e15 missing reference: Binder.;#5c:3;:U6;CHF;:H9af:8,E;:H9b0,E
Missing external geometry reference
Clear constraints to external geometry
<Sketch> SketchObject.cpp(9135): External geometry BaseSideWall#Sketch001.e15 missing reference: Binder.;#5c:3;:U6;CHF;:H9af:8,E;:H9b0,E
Missing external geometry reference
Clear constraints to external geometry
Traceback (most recent call last):
  File "/tmp/.mount_FreeCAZXcM2y/usr/Mod/Measure/MeasureCOM.py", line 114, in parseSelection
    obj.Element = (o, item["subName"])
    ^^^^^^^^^^^
<class 'ValueError'>: PropertyLinkSub does not support external object
Traceback (most recent call last):
  File "/tmp/.mount_FreeCAZXcM2y/usr/Mod/Measure/MeasureCOM.py", line 114, in parseSelection
    obj.Element = (o, item["subName"])
    ^^^^^^^^^^^
<class 'ValueError'>: PropertyLinkSub does not support external object
Traceback (most recent call last):
  File "/tmp/.mount_FreeCAZXcM2y/usr/Mod/Measure/MeasureCOM.py", line 114, in parseSelection
    obj.Element = (o, item["subName"])
    ^^^^^^^^^^^
<class 'ValueError'>: PropertyLinkSub does not support external object
Traceback (most recent call last):
  File "/tmp/.mount_FreeCAZXcM2y/usr/Mod/Measure/MeasureCOM.py", line 114, in parseSelection
    obj.Element = (o, item["subName"])
    ^^^^^^^^^^^
<class 'ValueError'>: PropertyLinkSub does not support external object
Program received signal SIGSEGV, Segmentation fault.
#0  /lib/x86_64-linux-gnu/libc.so.6(+0x45320) [0x72a15f645320]
#1  0x72a13d1d1f08 in Gui::TaskMeasure::saveObject() from /tmp/.mount_FreeCAZXcM2y/usr/lib/MeasureGui.so+0x98
#2  0x72a13d1d401a in Gui::TaskMeasure::apply() from /tmp/.mount_FreeCAZXcM2y/usr/lib/MeasureGui.so+0xa
#3  0x72a13d1d4166 in Gui::TaskMeasure::eventFilter(QObject*, QEvent*) from /tmp/.mount_FreeCAZXcM2y/usr/lib/MeasureGui.so+0xa6
#4  0x72a15fcb67ca in QCoreApplicationPrivate::sendThroughApplicationEventFilters(QObject*, QEvent*) from /tmp/.mount_FreeCAZXcM2y/usr/bin/../lib/libQt5Core.so.5+0x6a
#5  0x72a160b703e8 in QApplicationPrivate::notify_helper(QObject*, QEvent*) from /tmp/.mount_FreeCAZXcM2y/usr/bin/../lib/libQt5Widgets.so.5+0xb8
#6  0x72a16293e499 in Gui::GUIApplication::notify(QObject*, QEvent*) from /tmp/.mount_FreeCAZXcM2y/usr/bin/../lib/libFreeCADGui.so+0x99
#7  0x72a15fcb6b28 in QCoreApplication::notifyInternal2(QObject*, QEvent*) from /tmp/.mount_FreeCAZXcM2y/usr/bin/../lib/libQt5Core.so.5+0x108
#8  0x72a1603312fb in QGuiApplicationPrivate::processKeyEvent(QWindowSystemInterfacePrivate::KeyEvent*) from /tmp/.mount_FreeCAZXcM2y/usr/bin/../lib/libQt5Gui.so.5+0xdb
#9  0x72a16030b0bc in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) from /tmp/.mount_FreeCAZXcM2y/usr/bin/../lib/libQt5Gui.so.5+0xac
#10  /tmp/.mount_FreeCAZXcM2y/usr/plugins/platforms/../../lib/libQt5XcbQpa.so.5(+0x64bea) [0x72a15a494bea]
#11  /tmp/.mount_FreeCAZXcM2y/usr/bin/../lib/./libglib-2.0.so.0(+0x5bad7) [0x72a15e310ad7]
#12  /tmp/.mount_FreeCAZXcM2y/usr/bin/../lib/./libglib-2.0.so.0(+0x5eac7) [0x72a15e313ac7]
#13  /tmp/.mount_FreeCAZXcM2y/usr/bin/../lib/./libglib-2.0.so.0(g_main_context_iteration+0x2e) [0x72a15e3140ce]
#14  0x72a15fd0aa36 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) from /tmp/.mount_FreeCAZXcM2y/usr/bin/../lib/libQt5Core.so.5+0x66
#15  0x72a15fcb570b in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) from /tmp/.mount_FreeCAZXcM2y/usr/bin/../lib/libQt5Core.so.5+0x11b
#16  0x72a15fcbd212 in QCoreApplication::exec() from /tmp/.mount_FreeCAZXcM2y/usr/bin/../lib/libQt5Core.so.5+0x82
#17  0x72a16289f943 in Gui::Application::runApplication() from /tmp/.mount_FreeCAZXcM2y/usr/bin/../lib/libFreeCADGui.so+0x983
#18  /tmp/.mount_FreeCAZXcM2y/usr/bin/freecad(+0x7640) [0x62bbc651f640]
#19  /lib/x86_64-linux-gnu/libc.so.6(+0x2a1ca) [0x72a15f62a1ca]
#20  /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x8b) [0x72a15f62a28b]
#21  /tmp/.mount_FreeCAZXcM2y/usr/bin/freecad(+0x79a9) [0x62bbc651f9a9]`
```

Recovery worked great on re-start. Whew! :D

### Full version info

```shell
OS: Linux Mint 22 (X-Cinnamon/cinnamon)
Word size of FreeCAD: 64-bit
Version: 1.0.0RC1.38642 (Git) AppImage
Build type: Release
Branch: (HEAD detached at 1.0rc1)
Hash: 60a251354e203bdb924e2190fb9f5e18b48d7362
Python 3.11.9, Qt 5.15.13, Coin 4.0.3, Vtk 9.2.6, OCC 7.7.2
Locale: English/United States (en_US)
Installed mods: 
  * Help 1.0.3
  * A2plus 0.4.60n
```

### Subproject(s) affected?

Core

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by hlorus** [comment]
How do you even get to the dialog of the pad without closing the measure dialog, don't you get that popup which closes the measure dialog?

![image](https://github.com/user-attachments/assets/be86a295-fe26-4430-bf1d-df2280f0cca1)

**Comment by j8spencer** [comment]
> How do you even get to the dialog of the pad without closing the measure dialog, don't you get that popup which closes the measure dialog?

I do _not_ see that dialog. I can click in the model tree and edit in the properties window even while the measure dialog is open. I've noticed the same thing while a task dialog is open as well, and got myself into trouble a couple of times.

**Comment by j8spencer** [comment]
I could not create a crash in RC2 (or again in RC1). I *think* the specifics were that I had an edge created by a chamfer on a pad and I increased the pad length which invalidated the chamfer, but not sure. Anyway, no crash now.

RC2 seems to make sure that if you have a measurement shown and you click inside the properties box, the measurement disappears. The measurement task stays open, but I could not keep the measurement up and edit a property at the same time. No dialog box popped up, though. The measurement just disappeared as soon as I clicked in a text box.

I _was_ able to get some measurements stuck in the 3D view that I have no idea how to turn off except bail and come back.

![image](https://github.com/user-attachments/assets/aec88cb9-2100-4db9-aba1-95cfea845b45)
